sklearn/_fast_gradient_boosting/loss.pyx

```diff
@@ -0,0 +1,321 @@
+            # this is correct as long as we adjust the gradients. See e.g. LS
+            # loss
+            hessians = np.ones(shape=1, dtype=Y_DTYPE)
+        else:
+            hessians = np.empty(shape=shape, dtype=Y_DTYPE)
+
+        return gradients, hessians
+
+    @abstractmethod
+    def get_baseline_prediction(self, y_train, prediction_dim):
+        """Return initial predictions (before the first iteration).
+
+        Parameters
+        ----------
+        y_train : array-like, shape=(n_samples,)
+            The target training values.
+        prediction_dim : int
+            The dimension of one prediction: 1 for binary classification and
+            regression, n_classes for multiclass classification.
+
+        Returns
+        -------
+        baseline_prediction: float or array of shape (1, prediction_dim)
+            The baseline prediction.
+        """
+        pass
+
+    @abstractmethod
+    def update_gradients_and_hessians(self, gradients, hessians, y_true,
+                                      raw_predictions):
+        """Update gradients and hessians arrays, inplace.
+
+        The gradients (resp. hessians) are the first (resp. second) order
+        derivatives of the loss for each sample with respect to the
+        predictions of model, evaluated at iteration ``i - 1``.
+
+        Parameters
+        ----------
+        gradients : array-like, shape=(n_samples * prediction_dim)
+            The gradients (treated as OUT array).
+        hessians : array-like, shape=(n_samples * prediction_dim) or \
+            (1,)
+            The hessians (treated as OUT array).
+        y_true : array-like, shape=(n_samples,)
+            The true target values or each training sample.
+        raw_predictions : array-like, shape=(n_samples, prediction_dim)
+            The raw_predictions (i.e. values from the trees) of the tree
+            ensemble at iteration ``i - 1``.
+        """
+        pass
+
+
+class LeastSquares(BaseLoss):
+    """Least squares loss, for regression.
+
+    For a given sample x_i, least squares loss is defined as::
+
+        loss(x_i) = (y_true_i - raw_pred_i)**2
+    """
+
+    hessians_are_constant = True
+
+    def __call__(self, y_true, raw_predictions, average=True):
+        # shape (n_samples, 1) --> (n_samples,). reshape(-1) is more likely to
+        # return a view.
+        raw_predictions = raw_predictions.reshape(-1)
+        loss = np.power(y_true - raw_predictions, 2)
+        return loss.mean() if average else loss
+
+    def get_baseline_prediction(self, y_train, prediction_dim):
+        return np.mean(y_train).astype(Y_DTYPE)
+
+    def inverse_link_function(self, raw_predictions):
+        return raw_predictions
+
+    def update_gradients_and_hessians(self, gradients, hessians, y_true,
+                                      raw_predictions):
+        raw_predictions = raw_predictions.reshape(-1)
```

Comment:
I'm using `reshape` because it's more likely to return a view according to the numpy docs (see the [Note](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.ravel.html)). But I agree `ravel()` is more natural

Reply:
`.ravel()` also returns a view.